src/frontend/Util_ROM.cpp

```diff
@@ -461,6 +461,37 @@ int LoadROM(const char* file, int slot)
     }
 }
 
+u32* RomIcon(u8 data[512], u16 palette[16])
+{
+    // Get the 4-bit palette indexes
+    u8 indexes[1024];
+    for (int i = 0; i < 512; i++)
```

Comment:
you don't have to use three loops for this, the entire operation can be done at once.

Reply:
Yeah i didn't write this, this is Hydr8gon's code that you showed me.

Reply:
Hydr8gon is a derp